### ISSUE TITLE
Warns about "no newline at eof" even when code body disables linting

### ISSUE THREAD

**Issue by wardbell** [palantir/tslint#2057]
### Bug Report

- __TSLint version__: `3.15.1`
- __TypeScript version__: `2.0.10`
- __Running TSLint via__: (please choose one) CLI 

#### TypeScript code being linted

Posted first as [Angular issue #13924](https://github.com/angular/angular/issues/13924); the official response was to close the issue with the comment:  "file an issue with tslint".

Note in the following TS code generated by Angular's AOT compiler that
1. There is no newline at the end
1. The body contains `/* tslint:disable */`

```ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

export const styles:any[] = ['#speak[_ngcontent-%COMP%] { font-style: italic;  color: blue; }'];
```

FWIW,  if I induce a linting problem within the body of that code, that problem is ignored as expected. The problem seems to be confined to missing newline at EOF.

with `tslint.json` configuration:

```json
{
  "rules": {
    "class-name": true,
    "comment-format": [
      true,
      "check-space"
    ],
    "curly": true,
    "eofline": true,
    "forin": true,
    "indent": [
      true,
      "spaces"
    ],
    "label-position": true,
    "label-undefined": true,
    "max-line-length": [
      true,
      140
    ],
    "member-access": false,
    "no-arg": true,
    "no-bitwise": true,
    "no-console": [
      true,
      "debug",
      "info",
      "time",
      "timeEnd",
      "trace"
    ],
    "no-construct": true,
    "no-debugger": true,
    "no-duplicate-key": true,
    "no-duplicate-variable": true,
    "no-empty": false,
    "no-eval": true,
    "no-inferrable-types": true,
    "no-string-literal": false,
    "no-switch-case-fall-through": true,
    "no-trailing-whitespace": true,
    "no-unused-expression": true,
    "no-unused-variable": true,
    "no-unreachable": true,
    "no-use-before-declare": true,
    "no-var-keyword": true,
    "object-literal-sort-keys": false,
    "one-line": [
      true,
      "check-open-brace",
      "check-catch",
      "check-else",
      "check-whitespace"
    ],
    "quotemark": [
      true,
      "single"
    ],
    "radix": true,
    "semicolon": [
      "always"
    ],
    "triple-equals": [
      true,
      "allow-null-check"
    ],
    "typedef-whitespace": [
      true,
      {
        "call-signature": "nospace",
        "index-signature": "nospace",
        "parameter": "nospace",
        "property-declaration": "nospace",
        "variable-declaration": "nospace"
      }
    ],
    "variable-name": false,
    "whitespace": [
      true,
      "check-branch",
      "check-decl",
      "check-operator",
      "check-separator",
      "check-type"
    ],

    "component-selector-name": [true, "kebab-case"],
    "component-selector-type": [true, "element"],
    "use-input-property-decorator": true,
    "use-output-property-decorator": true,
    "use-life-cycle-interface": true,
    "use-pipe-transform-interface": true,
    "component-class-suffix": true,
    "directive-class-suffix": true,
    "import-destructuring-spacing": true
  }
}
```

#### Actual behavior

```
[23:29:55] [gulp-tslint] error inheritance/ts/aot/app/base.component.css.shim.ngstyle.ts[8, 97]: file should end with a newline
```

#### Expected behavior

No warning

**Comment by ajafff** [comment]
This is fixed by #1966 and should be included in tslint 4.3.0, although it's not in the changelog

**Comment by wardbell** [comment]
Thanks. Didn't see it in the change log. We'll see what it takes for us to upgrade to `4.3.x`
